src/main/java/com/hermesworld/ais/galapagos/devcerts/impl/DeveloperCertificateServiceImpl.java

```diff
@@ -119,6 +123,25 @@ public void init(KafkaCluster cluster) {
         return Optional.of(metadata);
     }
 
+    @Override
+    public CompletableFuture<Integer> clearExpiredDeveloperCertificatesOnAllClusters() {
+        CompletableFuture<Void> result = FutureUtil.noop();
+        AtomicInteger totalClearedDevCerts = new AtomicInteger();
+
+        for (KafkaCluster cluster : kafkaClusters.getEnvironments()) {
+            List<DevCertificateMetadata> expiredDevCerts = getRepository(cluster).getObjects().stream()
```

Comment:
Here, a check should be added that this cluster is certificate-based, otherwise, just continue.